gadget/ondisk.go

```diff
@@ -35,6 +35,15 @@ type OnDiskStructure struct {
 	// Node identifies the device node of the block device.
 	Node string
 
+	// StructureIndex is the index of the structure on the disk - this should be
+	// used instead of "Index" for an OnDiskStructure, "Index" comes from the
+	// embedded LaidOutStructure which is 0-based and does not have the same
+	// meaning. A LaidOutStructure's index position will include that of bare
+	// structures which will not show up as an OnDiskStructure, so the maximum
+	// OnDiskStructure.StructureIndex is not necessarily the same as the maximum
+	// LaidOutStructure.Index.
+	StructureIndex int
```

Comment:
this kind of changes are always delicate, ideally one would rename both and then maybe rename back one of the two. How can I be sure that all the relevant Index have turned into StructureIndex?. Also DiskIndex or OnDiskIndex would seem a better name for this one given the definition, no?

Reply:
broken out to https://github.com/snapcore/snapd/pull/11234